BEPs/BEP-530.md

```diff
@@ -0,0 +1,89 @@
+      - [4.1.3 Final Producer Filtering](#413-final-producer-filtering)
+    - [4.2 Algorithm Evaluation](#42-algorithm-evaluation)
+      - [4.2.1 Security Analysis](#421-security-analysis)
+      - [4.2.2 Fairness Assessment](#422-fairness-assessment)
+      - [4.2.3 Robustness Evaluation](#423-robustness-evaluation)
+      - [4.2.4 Simplicity Analysis](#424-simplicity-analysis)
+  - [5. Backwards Compatibility](#5-backwards-compatibility)
+  - [6. License](#6-license)
+
+# BEP-530: Improved Block Producer Selection Mechanism
+
+## 1. Summary
+This BEP proposes an improved block producer selection mechanism.
+
+## 2. Abstract
+This BEP introduces a new algorithm for selecting block producers, offering improvements in four key areas: security, fairness, robustness, and simplicity.
+
+## 3. Motivation
+The motivation is to increase the ratio of blocks produced by the "in-turn" validator to enhance network stability. Additionally, it aims to improve the predictability of backup validators at each block height to support optimized transaction and block propagation.
+
+## 4. Specification
+### 4.1 Algorithm Overview
+#### 4.1.1 Validator Set Retrieval
+Each epoch involves a change in the validator set. After retrieving the new set from the contract, we use `epochNumber (blockNumber / EpochLength)` as the seed to randomly shuffle the validator set. Validators are assigned an index within the range `[0, len(validators))`, and this index remains unchanged for the duration of the epoch.
+
+#### 4.1.2 Determining Turn Order
+At any given block height, one validator is designated as the "in-turn" validator with the highest priority to produce a block, with a block difficulty of 2. Additionally, multiple backup validators are assigned with a block difficulty of 1, and their blocks are broadcasted with a delay. The index of the in-turn validator at block height     `blockNumber` is determined as follows:
+```Go
+// turnLength specifies how many consecutive blocks a validator can produce
+inturnIndex := blockNumber / uint64(turnLength) % uint64(len(validators))
+```
+
+Backup validators are selected by reversing the order of the validators prior to the `inturnIndex`. The block producer sequence, including both in-turn and backup validators, is shown as follows:
+```Go
+// Reverse round	
+for order := 0; order < len(validators); order++ {
+			log.Debug("block producing", "order", order,
+				"validator index", (inturnIndex+len(validators)-order)%len(validators))
+}
+```
+
+#### 4.1.3 Final Producer Filtering
+To prevent chain splits, Parlia continuously records the recent block producers within a given length, known as `MinerHistory`. Once these validators reach a predefined block production threshold, they are no longer eligible to produce blocks at the current height. This BEP proposes using the latest finalized block to clean `MinerHistory`, removing producers who have produced blocks up to and including the latest finalized block.
```

Comment:
Could you help to explain why it can prevent chain splits? Another question is does that undermine fairness? Maybe some validators will produce more blocks?